### PR TITLE
09 GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by ryanbirdi97**

### CODE REVIEW

**Review by jimstev2447 — CHANGES_REQUESTED** [review]
I want you to think about the case in which there is an article that has no comments, I would expect you to return a 200 and an empty array, this is currently not being considered

File: __tests__/app.test.js [hunk 1]
```diff
@@ -192,11 +192,59 @@ describe("GET /api/articles", () => {
...
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("responds with a array of comment objects of given id", () => {
+    const articleID = 5;
+    return request(app)
+      .get(`/api/articles/${articleID}/comments`)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toBeInstanceOf(Array);
+        body.comments.forEach((comment) => {
```
Comment (two-triangles) [inline __tests__/app.test.js:213]: if we are using forEach - you should be testing the length of the array of comments beforehand.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -192,11 +192,59 @@ describe("GET /api/articles", () => {
...
+        expect(body.msg).toBe(`Bad Request`);
+      });
+  });
+  test("200: if article has no comments returns an empty array", async () => {
+    const articleID = 2;
+    return request(app)
+      .get(`/api/articles/${articleID}/comments`)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toBeInstanceOf(Array);
```
Comment (two-triangles) [inline __tests__/app.test.js:249]: We should be checking what the array looks like as well - an empty array

**Comment by two-triangles** [comment]
Just one comment :)

**Review by CatHoang — APPROVED** [review]
Looks good Ryan, feel free to merge!

**Pull request merged**